scapy/contrib/isotp.py

```diff
@@ -664,17 +664,11 @@ def run(self):
         ins = self.socket
 
         while 1:
-            pkts = ins.sniff(timeout=1, count=1)
-            if len(pkts) == 1:
-                cf = pkts[0]
-            else:
-                cf = None
+            ins.sniff(store=False, timeout=1,
```

Comment:
Do you think you could remove the timeout, then the whole while loop, to simply use sniff ?

The stop_filter will take care of exiting

Reply:
As far as I understood the stop_filter, an exit can only happen, if a message is received. 
If this is the case, I can not close the socket, if no further message is received. Sending a "fake message" just to trigger the execution of `stop_filter` inside `close` won't work, since I can not inject a message into the CAN socket. 
Do you see a possibility to trigger the execution of stop_filter without a receive of a message?

Reply:
Correct... I had a prototype https://github.com/secdev/scapy/pull/1523 of a such implementation, but that wasnt clean enough (now it's outdated).

I'll come back to you if i get any additional options available